src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -14,12 +14,310 @@
+    return true;
+}
+
+/**
+ * Given a nuget version, detect if a serviced hostpolicy is available at
+ *   platform servicing location.
+ */
+bool hostpolicy_exists_in_svc(const pal::string_t& version, pal::string_t* resolved_dir)
+{
+    if (version.empty())
+    {
+        return false;
+    }
+
+    pal::string_t svc_dir;
+    pal::get_default_servicing_directory(&svc_dir);
+    append_path(&svc_dir, _X("pkgs"));
+    return to_hostpolicy_package_dir(svc_dir, version, resolved_dir);
+}
+
+/**
+ * Given path to app binary, say app.dll, retrieve the app.deps.json.
+ */
+pal::string_t get_deps_from_app_binary(const pal::string_t& app)
+{
+    assert(app.find(DIR_SEPARATOR) != pal::string_t::npos);
+    assert(ends_with(app, _X(".dll"), false) || ends_with(app, _X(".exe"), false));
+
+    // First append directory.
+    pal::string_t deps_file;
+    deps_file.assign(get_directory(app));
+    deps_file.push_back(DIR_SEPARATOR);
+
+    // Then the app name and the file extension
+    pal::string_t app_name = get_filename(app);
+    deps_file.append(app_name, 0, app_name.find_last_of(_X(".")));
+    deps_file.append(_X(".deps.json"));
+    return deps_file;
+}
+
+/**
+ * Given a version and probing paths, find if package layout
+ *    directory containing hostpolicy exists.
+ */
+bool resolve_hostpolicy_dir_from_probe_paths(const pal::string_t& version, const std::vector<pal::string_t>& probe_realpaths, pal::string_t* candidate)
+{
+    if (probe_realpaths.empty() || version.empty())
+    {
+        return false;
+    }
+
+    // Check if the package relative directory containing hostpolicy exists.
+    for (const auto& probe_path : probe_realpaths)
+    {
+        trace::verbose(_X("Considering %s to probe for %s"), probe_path.c_str(), LIBHOSTPOLICY_NAME);
+        if (to_hostpolicy_package_dir(probe_path, version, candidate))
+        {
+            return true;
+        }
+    }
+
+    // Print detailed message about the file not found in the probe paths.
+    trace::error(_X("Could not find required library %s in %d probing paths:"),
+        LIBHOSTPOLICY_NAME, probe_realpaths.size());
+    for (const auto& path : probe_realpaths)
+    {
+        trace::error(_X("  %s"), path.c_str());
+    }
+    return false;
+}
+
+/**
+ * Given FX location, app dll and specified --depsfile, return deps that contains hostpolicy.dll
+ */
+pal::string_t get_deps_file(
+    const pal::string_t& fx_dir,
+    const pal::string_t& app_candidate,
+    const pal::string_t& deps_file,
+    const runtime_config_t& config)
+{
+    if (config.get_portable())
+    {
+        // Portable app's hostpolicy is resolved from FX deps
+        return fx_dir + DIR_SEPARATOR + config.get_fx_name() + _X(".deps.json");
+    }
+    else
+    {
+        // Standalone app's hostpolicy is from specified deps or from app deps.
+        return !deps_file.empty() ? deps_file : get_deps_from_app_binary(app_candidate);
```

Comment:
Please rename "deps_file" to "specified_deps_file" to indicate that it was passed at the command line by the user.